### ISSUE TITLE
Override Global Dark List

### ISSUE THREAD

**Issue by phr00t** [darkreader/darkreader#824]
The Global Dark List cannot be changed or overwritten by end users and it locks out any configuration for those pages. This may be by design, but it is quite restrictive. For example, Discord's "dark" mode is still relatively bright when you are used to a black background, but I can't use this Add-On to make any changes to the Discord website. I feel like I should be able to apply custom settings to a website if I choose, overwriting the Global Dark List.

**Comment by markini** [comment]
That would be nice. 
Currently I have problems with imgur because it is on the global dark list.
If you access an gifv like this however:
![imgur_hmm](https://user-images.githubusercontent.com/1747400/48293065-8a788800-e47d-11e8-966d-62dc74137bbe.png)

**Comment by doubledgedboard** [comment]
Similar issues on https://www.lolesports.com/en_US/articles

**Comment by alexanderby** [comment]
Currently I removed Imgur and Lolesports from this list.

**Comment by satvikpendem** [comment]
Why can't we configure our own dark list? Or manually override it, instead of you having to add or remove domains every time there is a problem?

**Comment by elderlabs** [comment]
Was attempting to read this article and was blinded
`https://www.pluralsight.com/blog/it-ops/linux-hardening-secure-server-checklist`

**Comment by bdrelling** [comment]
OpenCritic is also blinding: https://opencritic.com/

We need to be able to manually configure it. This has gone on for way too long.

**Comment by Horizonistic** [comment]
https://ffmpeg.org/ is on the global list which looks fine until you go to a page like this:
https://trac.ffmpeg.org/wiki/Encode/H.264

The fact that I can't override this is a bit silly.

**Comment by VelocityIsNotSpeed** [comment]
http://www.hardcoregaming101.net/ is in the global dark list but it is blindingly white.

Being totally unable to do anything about these sites with this extension is insane. It's otherwise the best "dark everywhere" extension i have tried, but this design flaw alone is huge.

**Comment by shatteredhalo** [comment]
Adding my vote here for user-customizable global dark lists!

**Comment by Gunni** [comment]
Super simple solution is to have this list define the _default_ for that specific page, but if the user clicks invert then that should take precedence over the default.

Site that is in neither list:
- Default: invert
- Global list: nothing
- User selections: nothing
- result: invert

Site in the global list but not user list:
- Default: invert
- Global list: do not invert
- User selections: nothing
- result: do not invert

Site in the global list and user list:
- Default: invert
- Global list: do not invert
- User selections: invert
- result: invert

Basically, do what the user wants, not what _you_ think is correct.

**Comment by Gunni** [comment]
Here, I made a truth table, true = invert, false = do not invert.

Basically the result is `user OR global OR default`.

default | global | user | result
-- | -- | -- | --
FALSE | FALSE | FALSE | FALSE
TRUE | TRUE | TRUE | TRUE
FALSE | TRUE | FALSE | TRUE
TRUE | FALSE | TRUE | TRUE
FALSE | FALSE | TRUE | TRUE
TRUE | TRUE | FALSE | TRUE
FALSE | TRUE | TRUE | TRUE
TRUE | FALSE | FALSE | TRUE

**Comment by arnobpl** [comment]
Please remove _dpreview.com_ from the global list, as the website is not dark. Also, please consider a better solution such as the one suggested by @Gunni

**Comment by ngklingler** [comment]
app.getpocket.com is also in the global darklist but is not at all dark themed. Would be nice to have a way to override this.

**Comment by zetapulse** [comment]
steamdb.info barely qualifies as dark and then you click on a game and it's all white fml.
Look Alex, you have a great extension here that's got a lot going for it, but if your intention behind this global dark list was to make the public more aware of making computer displays easier on the eyes, it's not gonna happen when you've got users resorting to using alternative extensions to get around arbitrary functions the likes of which Apple would force on its customers.

Let the global dark list disable Dark Reader, but with the option for the user to enable it in case they prefer the uniformity your extension offers or in case a website changes or has some random white pages.

**Comment by ei14** [comment]
i3wm.org has a dark home page, but many directories are white, and almost every single subdomain of this website is white.

**Comment by lodewykk** [comment]
How about a "**_Force_** on" button, you could use?

**Comment by aksh1618** [comment]
Totally frustrated by the inablility to use this otherwise great extension in case of a whole domain being in some "list"! Just let me decide if I want to use the extension on some page instead of hindering the functionality without any kind of an override. A big design flaw imho. (Was blinded by [this](https://www.pluralsight.com/guides/data-stream-management-with-apache-kafka-streams))

**Comment by Raylier** [comment]
Went to Pluralsight, is there any update on the user configurable list? I don't like having my eyes burned at 2AM

**Comment by PicchiarelloMatto** [comment]
> Anche app.getpocket.com è nella darklist globale ma non è affatto a tema dark. Sarebbe bello avere un modo per ignorare questo.

Actually app.getpocket.com has an option to activate the dark theme, I think this is why it was put on the darklist, but the setting is memorized in the cookies, so if you delete them you return to the  default theme unfortunately. I have also tried to add app.getpocket.com to Chrome exclusions, but only the login is memorized, and the dark theme is still lost.

**Comment by Kezxo** [comment]
What if you want dark website brighter by using **Light Mode** for whatever reason? You can do that on any website besides the one on the dark list.

The known dark websites are disallowed from using light mode, how does that make sense? Am I missing something?

**Comment by kittydoor** [comment]
https://faq.i3wm.org is also blinding.

Please allow an override of the global dark list, manually managing the list will never be as good as just letting us manage it...

**Comment by Committing** [comment]
Hackforums is in the global dark list and I cannot change any of it.

Please remove this RUBBISH global list that is hidden on some remote server or allow us to edit it.

I'll be uninstalling this plugin now

**Comment by alexanderby** [comment]
Until it is fixed you can edit the [dark-sites.config](https://github.com/darkreader/darkreader/blob/master/src/config/dark-sites.config) file, click Edit, remove bright website, submit a Pull Request.

**Comment by Mad-Professor** [comment]
Why the hell can I not changes this locally? Why are you forcing us to refrence a global list. Defeats the point of controlling the viewing experiance. Change this so users still have control.

I'm pissed off because I get on twitch and get blasted by white bright screen, especially when I just got done working in the dark. I'm not a registered user so I can't change to dark mode. Why the hell would you do this?
Removing plugin, going to find something else, that doesn't force this kind of bullshit.

**Comment by Gunni** [comment]
Controversial(?) idea:

Empty the [dark-sites.config](https://github.com/darkreader/darkreader/blob/master/src/config/dark-sites.config) until @alexanderby, or someone else has a better fix for this usability issue.

**Comment by alexanderby** [comment]
Well, there was a [Pull Request](https://github.com/darkreader/darkreader/pull/1584) recently that added Twitch to the list. It was dark by default when I checked.

I've got an idea how to add a simple patch for it without changing the UI, please wait for a day or two.

@Gunni Nice idea :) I would suggest you to find out where Dark Reader is installed on your machine and clean this list there.

**Comment by Mad-Professor** [comment]
@alexanderby 
The issue is "twitch.tv" domain is default dark, but sub domain "clips.twitch.tv" is not and there's no way to change it via twitch unless registered, especially on "clips.twitch.tv". The hamburger menu is removed (design oversight or intentional?) and by adding it to the darkreader darklist you effectively removed control from the user. There's no over-ride, I had to clone master and basically empty the list (ctrl-a then del followed ctrl-s) then load unpack extension in chromium for the time being, that was after perusing for an alternative then coming to the conclusion that there really isn't alternative, or at least one that is currently being maintain or doesn't come close in features to darkreader.
In the end I lost 4 hours and I wanted to punch my monitor everytime I hit page of white.

**Comment by alexanderby** [comment]
I've made a fix, but I will publish it tomorrow since it touches sensitive settings areas and can break something for other users.

**Comment by GoldAnanas** [comment]
remove twitch.tv from dark list please... or at least do something to override the list. Thanks ! (I love your extension)

**Comment by Lothrik** [comment]
It's not just clips.twitch.tv, all of it is blinding white now. Please remove it from the global dark list or give users a way to override it, thanks!

**Comment by alexanderby** [comment]
It should be now possible to toggle websites in Dark List since version 4.8.3.

**Comment by asd-collab** [comment]
get blinded by https://source.joinmastodon.org/mastodon/docs, can't be changed as its on the 'global dark list'. its all freaking white!

extension Version 4.8.4
`on latest Firefox stable`

only God knows what other site is also on this stupid, useless list.

can't 'Apply settings to current website only' and change filter to any setting that works.
----

could have been fixed a month after Nov 6 2018 ( https://github.com/darkreader/darkreader/issues/824#issue-377666539 )

## **just remove the `global Dark list`.** that's the fix.

---

asd-collab

**Comment by sanjaylord** [comment]
Hello, would it be possible to remove https://forum.snahp.it/ from the Global Dark List. I've already made the edit to the dark-sites.config file and submitted a Pull Request for it. Thanks.

**Comment by vaaski** [comment]
after being frustrated that [status.discordapp.com](https://status.discordapp.com/) was blinding me without me being able to toggle it with the keyboard shortcut, I found out that you can manually add the site to the site list to override the global dark list, like so:

![image](https://user-images.githubusercontent.com/17879327/76804258-20dc5c80-67dc-11ea-99b0-4a84ee918d07.png)

**Comment by Committing** [comment]
Thank you @vaaski

**Comment by F8ER** [comment]
> after being frustrated that [status.discordapp.com](https://status.discordapp.com/) was blinding me without me being able to toggle it with the keyboard shortcut, I found out that you can manually add the site to the site list to override the global dark list

Doesn't work if you have this list filled already and set to `Invert listed only`.

**Comment by Gusted** [comment]
> Doesn't work if you have this list filled already and set to `Invert listed only`.

`*.discord.com` just darkening every subdomain that isn't dark(of discord) and discord itself isn't being darken.

**Comment by F8ER** [comment]
> > Doesn't work if you have this list filled already and set to `Invert listed only`.
> 
> `*.discord.com` just darkening every subdomain that isn't dark(of discord) and discord itself isn't being darken.

I mean, there's no way to exclude particular Global Listed website if this list is already being used and set to `Invert listed only`.

**Comment by sanjaylord** [comment]
> > > Doesn't work if you have this list filled already and set to `Invert listed only`.
> > 
> > 
> > `*.discord.com` just darkening every subdomain that isn't dark(of discord) and discord itself isn't being darken.
> 
> I mean, there's no way to exclude particular Global Listed website if this list is already being used and set to `Invert listed only` or am I missing something?

![76804258-20dc5c80-67dc-11ea-99b0-4a84ee918d07_cropped](https://user-images.githubusercontent.com/60788875/89621967-e620b100-d8af-11ea-87b8-0a25be962c62.png)

Just press the ✔ or ❎ respectively next to the site name to manually ignore a site from "Global Dark List".

**Comment by F8ER** [comment]
> Just press the ✔ or ❎ respectively next to the site name to manually ignore a site from "Global Dark List".

![](https://i.imgur.com/Tpn4tmQ.gif)

**Comment by sanjaylord** [comment]
> > Just press the ✔ or ❎ respectively next to the site name to manually ignore a site from "Global Dark List".
> 
> <img alt="" width="300px" src="https://camo.githubusercontent.com/2bf547861a4177cfa2eb090f4ae2bff4b9def3b5/68747470733a2f2f692e696d6775722e636f6d2f54706e34746d512e676966">
> 
> Also, the Discord might refer to `discord.com` from `discordapp.com`.

Just tried on the very same page (status.discord.com) and the toggle works as expected for me. Try making sure you don't have the domain already added to either your "Invert Only" or "Not Invert" list.

If that doesn't work then try resetting/re-installing the extension.

**Comment by Gusted** [comment]
> Try making sure you don't have the domain already added to either your "Invert Only" or "Not Invert" list.

Main TLD domain so discord.com this is another issue that should be fixed by #373

**Comment by orrinpants** [comment]
I wanted the opposite so I guess I'll be fine to add it to the "Local dark list".

**Comment by jaskapaari** [comment]
I'd like to point out, that these global lists don't work as expected for other but Dynamic filter mode. I use Filter theme generation mode, and for example i.imgur.com images are un-viewable (4.9.52), and I can't override it by adding it to non-invert list, nor clicking the toggle, because the site's on "Global Dark List".
Maybe make this global list locally overridable, or make it only apply to Dynamic mode.

**Comment by Powsin** [comment]
Mobalytic's TFT Team Builder is unusable on firefox. You can't see the grid lines for you to place your team into.

https://app.mobalytics.gg/tft/comp-builder

Please remove it from the global list.

**Comment by Gliktch** [comment]
> after being frustrated that [status.discordapp.com](https://status.discordapp.com/) was blinding me without me being able to toggle it with the keyboard shortcut, I found out that you can manually add the site to the site list to override the global dark list, like so:
> 
> ![image](https://user-images.githubusercontent.com/17879327/76804258-20dc5c80-67dc-11ea-99b0-4a84ee918d07.png)

Maybe this has been fixed since your post, but I've been able to use the shortcut key (Alt+Shift+A) on a global-dark-listed site, to now carry out this adding to the invert list for me, without extra steps. 🤓👍️
